### ISSUE TITLE
validation 및 reproducibility 관련 문의

### ISSUE THREAD

**Issue by bonob12** [LISTatSNU/FastMRI_challenge#340]
안녕하세요 EJ 팀의 주의진입니다!

다름이 아니라 seed를 고정해놓아도 처음 몇 백번의 iteration에서 loss 동일하게 찍히지만 

이후 iteration이 늘어날 수록 소숫점 5~6번째 자리에서 조금씩 오차가 생기기 시작해서 전체적인 loss가 

미세하게 다르게 찍히는 것을 관찰하였습니다. 부동소수점 오차에 의한 것 같은데 이것이 문제가 없을지 여쭤보고 싶습니다.

또한 학습 과정에서  validation 없이 데이터를 모두 train에 쓰려고 하는데 

재현성 검증을 위해 validation loss log 대신 train loss log를 저장하여 제출해도 괜찮을까요?

감사합니다.

**Comment by B0neh3ad** [comment]
@bonob12 안녕하세요,

말씀하신 수준의 오차는 허용 가능합니다. 다만, 부동소수점 오차는 deterministic하므로 시행에 따라 다르게 나타날 이유는 없어보입니다. 모든 seed가 고정이 되었는지 재차 확인해주시면 감사하겠습니다. (baseline의 `seed_fix()` 함수로 고정되지 않는 seed가 존재한다면 별도로 고정해주시기 바랍니다.)

val data 없이 모두 train에 사용하는 경우, 말씀하신 대로 train loss log를 제출하셔도 됩니다. 다만 추후 조교들이 검증 시 log 파일을 생성 및 확인할 수 있도록 readme에 관련 내용을 기재해주시기 바랍니다. 이에 대해서는 final submission 관련 영상 및 자료에서 재차 안내드리겠습니다.

감사합니다.

**Comment by bonob12** [comment]
seed_fix() 함수에 torch.use_deterministic_algorithms(True)를 추가했더니 해결되었습니다!

감사합니다.
